stm/stm.ml

```diff
@@ -496,21 +494,14 @@ end = struct (* {{{ *)
 
   let checkout_shallowest_proof_branch () =
     if List.mem edit_branch (Vcs_.branches !vcs) then begin
-      checkout edit_branch;
-      match get_branch edit_branch with
-      | { kind = `Edit (mode, _,_,_,_) } -> Proof_global.activate_proof_mode mode
-      | _ -> assert false
+      checkout edit_branch
     end else
       let pl = proof_nesting () in
       try
-        let branch, mode = match Vcs_aux.find_proof_at_depth !vcs pl with
-          | h, { Vcs_.kind = `Proof (m, _) } -> h, m | _ -> assert false in
-        checkout branch;
-        stm_prerr_endline (fun () -> "mode:" ^ mode);
-        Proof_global.activate_proof_mode mode
+        let branch = fst @@ Vcs_aux.find_proof_at_depth !vcs pl in
```

Comment:
$, or @@, is nice IMO. It is like opening `(` and saying "until the end of the line".
It was invented quite some time ago in Haskell.
I quite don't get why `@@` was preferred to `$`, but still..

`|>` has the nice property of putting the subject upfront, like the `|` of unix.

IMO, depending on the situation and the stack depth one or the other may work better.
Here, for example, the stack is 1 slot and contains `fst`, so I can stand it ;-)